File: rd-net/RdFramework/ExtCreationInfo.cs
using JetBrains.Diagnostics;
using JetBrains.Rd.Base;

namespace JetBrains.Rd
{
  public readonly struct ExtCreationInfo
  {
    public readonly RName Name;
    public readonly RdId? Id;
    public readonly long Hash;
    public readonly RdExtBase? Ext;

    public ExtCreationInfo(RName name, RdId? id, long hash, RdExtBase? ext)
    {
      Name = name;
      Id = id;
      Hash = hash;
      Ext = ext;
    }
  }

  public readonly struct ExtCreationInfoEx
  {
    public readonly ExtCreationInfo Info;
    public readonly bool IsLocal;

    public ExtCreationInfoEx(ExtCreationInfo info, bool isLocal)
    {
      Info = info;
      IsLocal = isLocal;
    }
  }
}

File: rd-kt/rd-framework/src/main/kotlin/com/jetbrains/rd/framework/ExtCreationInfo.kt
package com.jetbrains.rd.framework

import com.jetbrains.rd.framework.base.RdExtBase
import com.jetbrains.rd.util.string.RName

data class ExtCreationInfo(
    val rName: RName,
    val rdId: RdId?,
    val hash: Long,
    val ext: RdExtBase?
)

data class ExtCreationInfoEx(
    val info: ExtCreationInfo,
    val isLocal: Boolean
)
